### PR TITLE
[Updated: should be fixed. please squash commits to latest when merging] Component endpoints

### PR DESCRIPTION
**Pull request by McCloudFace**
- [x] Add api endpoint for each of my widgets `Banner` and `TrailDescription`

* When running the `trail-service`, `trail-photos-service`, and `reviews-service` locally, requests to these endpoints return responses shaped for each component state:
    - `http://localhost:3001/1/banner`
    - `http://localhost:3001/1/trailDescription`

@MengRS or @weber93 - if you could review & merge when you get a chance? Thanks!

### CODE REVIEW

**Comment by MengRS** [comment]
@McCloudFace, I noticed nodemon is missing in your package.json when I did a fresh install.  Last I asked, I believe Leslie wanted us to add all node packages into package.json (including nodemon).

**Comment by MengRS** [comment]
@McCloudFace, review complete - see notes below.


    - `http://localhost:3001/1/banner` 
Comment: Received below error
image

    - `http://localhost:3001/1/trailDescription`
Comment: Looks like the below... not sure what the actual shape is suppose to look like for this particular api endpoint (even as I am looking at your readme and your example-data.json file... I believe some fields are missing... e.g. `difficulty`, `origin`, etc. ... if it's as designed, let me know / you may want to consider editing the docs)
image

**Comment by McCloudFace** [comment]
Hey @MengRS - thanks very much for those call-outs, I appreciate the detailed feedback! I probably messed something up last night, it was pretty late for me :) For some weird reason, the `axios` package is listed in devDependencies on this branch, but node couldn't find it this morning. I ran `npm i` to re-install it (also added `nodemon`, thanks!). After that, I started up the services and `http://localhost:3001/1/banner` appears to be responding as expected.

I had an OH w Leslie and I'm making these two new endpoints specifically for my widgets. They return the data pre-shaped for the component state of each widgety widg. These won't be consumed by other services, just my widgets. I will make notes on the readme for sure. If you pull latest, then `npm i` and start up the services, I believe you should see what I'm seeing for `banner`? Thanks!!

**Comment by MengRS** [comment]
Hi @McCloudFace, I re-cloned your branch, npm installed, adjusted .env and launched up all services... and got the same `totalReviews` issue for `http://localhost:3001/1/banner` as indicated above ... click [here](https://imgur.com/a/LfHjTNN) for my experience....

**Comment by cjm771** [comment]
yeah i get same thing , this seems to be response for bannerstats `{ rank: '#11 ranked trail out of 20',
  heroUrl: 'https://images.unsplash.com/photo-1533412992015-1ddf680fe6dd?ixlib=rb-1.2.1&q=80&fm=jpg&crop=entropy&cs=tinysrgb&w=700&h=520&fit=crop&ixid=eyJhcHBfaWQiOjF9',
  count: '50' }`.

 I would say you should probably promise.all in your service helper before, this looks like an async issue.

File: server/serviceHelpers.js [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+
+  axios.get(trailStats)
+    .then((response) => {
+      result.stats = response.data;
+    })
+    .catch((error) => {
+      console.error('axios.get trailStats: ', error);
+    });
+
+  axios.get(trailRank)
```
Comment (cjm771) [inline server/serviceHelpers.js]: are each of these async? consider either using async / await or a Promise.all to ensure you get all data before resolving this promise / running your callback. @MengRS  @McCloudFace

**Comment by McCloudFace** [comment]
Thanks @MengRS and @cjm771 - appreciate it! I'll have a look and get back to you

**Comment by McCloudFace** [comment]
scravajipt

Thanks @MengRS & @cjm771 - it was broked. I think it's all betta now. Axios has a "make all yer calls then do stuff when yer done" syntax, which [shocker] I missed in the docs the 1st time around.

Should be functioning as-expected now.  Thanks again, sorry for the troubles.

**Comment by MengRS** [comment]
Hey @McCloudFace, could be me but I get this now :( 
image

**Pull request merged**